packages/react-components/react-provider/src/components/FluentProvider/useFluentProviderThemeStyleTag.ts

```diff
@@ -65,6 +68,13 @@ export const useFluentProviderThemeStyleTag = (options: Pick<FluentProviderState
   const rule = `.${styleTagId} { ${cssVarsAsString} }`;
 
   useInsertionEffect(() => {
+    // The style element could already have been created during SSR - no need to recreate it
+    const ssrStyleElement = targetDocument?.getElementById(styleTagId) as HTMLStyleElement;
```

Comment:
Let's simplify logic here:

```js
    const ssrStyleElement = targetDocument?.getElementById(styleTagId);

    if (ssrStyleElement) {
      styleTag.current = ssrStyleElement as HTMLStyleElement;
    } else {
      styleTag.current = createStyleTag(targetDocument, { ...styleElementAttributes, id: styleTagId });

      if (styleTag.current) {
        insertSheet(styleTag.current, rule);
      }
    }

    return () => {
      styleTag.current?.remove();
    };
```

I also moved type cast, so we will not loose a check for `undefined`